cms/search/publishers.py

```diff
@@ -0,0 +1,202 @@
+import json
+import logging
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Optional
+
+from django.conf import settings
+from kafka import KafkaProducer
+
+logger = logging.getLogger(__name__)
+
+
+if TYPE_CHECKING:
+    from kafka.producer.future import RecordMetadata
+    from wagtail.models import Page
+
+
+class BasePublisher(ABC):
+    """BasePublisher defines shared functionalities, such as how to build a message
+    for created/updated or deleted events. Each subclass only needs to define how
+    to actually publish (i.e., send) the built message.
+
+    dp-search-data-extractor spec link:
+    https://github.com/ONSdigital/dp-search-data-extractor/blob/develop/specification.yml#L53
+    dp-search-data-importer spec link:
+    https://github.com/ONSdigital/dp-search-data-importer/blob/30fb507e90f2cf1974ec0ca43bb0466307e2f112/specification.yml#L186
+    contract: https://github.com/ONSdigital/dis-search-upstream-stub/blob/main/docs/contract/resource_metadata.yml
+    """
+
+    def publish_created_or_updated(self, page: "Page") -> None:
+        """Build the message for the created/updated event.
+        Delegate sending to the subclass's _publish_to_service().
+        """
+        topic = self.get_topic_created_or_updated()
+        message = self._construct_message_for_create_update(page)
+        logger.info(
+            "BasePublisher: About to publish created/updated message=%s to topic=%s",
```

Comment:
```
BasePublisher: About to publish created/updated message={...}
LogPublisher: topic=log-created-or-updated message={...}
```

Doesn't flow. IMHO, the log messages should be consistent and specify the actual publisher. ie. if I am using `LogPublisher` then all log messages should state that.

Additionally, KafkaPublisher._publish_to_service does:

```
KafkaPublisher: Publishing to topic=%s, message=%s
KafkaPublisher: Publish result for topic %s: %s
```

so we end up with 

```
BasePublisher: About to publish created/updated message={...}
KafkaPublisher: Publishing to topic=%s, message=%s
KafkaPublisher: Publish result for topic %s: %s
```